drivers/hmis/app/graphql/types/forms/pick_list_option.rb

```diff
@@ -447,5 +449,11 @@ def self.available_units_for_enrollment(project, household_id: nil)
           }
         end
     end
+
+    def self.assessment_types_for_project(project)
+      project.custom_assessments.joins(:form_processor, :definition).pluck(:identifier, :title).uniq.map do |id, title|
```

Comment:
One downside to this change is that, since the pick list now only returns options if a custom assessment exists with that identifier for this project, then the pick list *won't* return options for assessments that are enabled for this project, but none exist yet. 

Maybe this is acceptable, and even positive? But I did think it may confuse users, so let me know if I should seek a different approach.

Reply:
I don't think that's ideal UX, and it will be slow to look through all those assessments.  This also needs to work if project is _not_ specified, because the Client assessments table has a type filter too.

Below is a suggestion. It's a little odd to combine the "roles" (eg `INTAKE`) with the identifiers (eg `housing_needs_assessment`), but I think we need to do that in order to get the desired behavior. The "Intake" option should show _all_ Intakes, regardless of what form they used. 

Thoughts? Happy to discuss more at standup

```
      # get all form rules for custom assessments (active and inactive)
      scope = Hmis::Form::Instance.with_role(:CUSTOM_ASSESSMENT)

      # filter down to rules that match this project, if project is specified
      scope = scope.filter { |fi| fi.project_match(project) } if project

      # { code: definition.identifier, label: definition.title }
      custom_options = scope.map(&:to_pick_list_option).uniq.sort_by { |opt| opt[:label] }

      hud_options = Hmis::Form::Definition::FORM_DATA_COLLECTION_STAGES.excluding(:CUSTOM_ASSESSMENT).keys.
        map { |k| { code: k.to_s, label: k.to_s.humanize } }

      hud_options + custom_optionss

```

Reply:
Yes this makes sense! Thanks for setting me straight.

After implementing this change I was wondering if it's worthwhile to create a new enum type in graphql, or should this input just be a list of strings?
- There is already an `AssessmentType` enum, from [HUD](https://github.com/greenriver/hmis-warehouse/blob/656e7c43e3a3e3d061f082d6eaf34296361d8c29/lib/util/concerns/hud_lists_2024.rb#L1121-L1123), so the names would conflict. We could name it something like `AssessmentRoleOrIdentifier`
- Could add a new `use_enum_with_same_key` on `Hmis::Form::Definition` that returns all of the identifiers for forms with role CUSTOM_ASSESSMENT
- maybe would need to make changes to `Hmis::FieldMap` to enable functionality like merging, and excluding specific keys?

Reply:
It can't be an enum because the form `identifier`s are dynamic basic on what's in the database. I think what you have with strings makes sense.
Maybe we should call it `assessment_name`?

Reply:
I'm good with switching to calling it "assessment name." I can make that change on the frontend too.

Reply:
Although, I'm seeing we use the language of 'Assessment Type' (rather than 'Name') on the frontend in a user-facing way. I guess it's two slightly different concepts,
- Assessment Type could be Intake, Exit ... Custom and corresponds to hud data collection stage
- Assessment Name could be Intake, Exit, but also SPDAT, "Martha's Custom Assessment," etc.

Do you think users of the app will understand this distinction? Maybe it doesn't really matter?

Reply:
I think either one is fine. They are both better than just "type". If we get feedback around wording later, we can override the label on the frontend.